pkgs/shelf_router/lib/src/router.dart

```diff
@@ -190,6 +190,21 @@ class Router {
     return _notFoundHandler(request);
   }
 
+  /// Get URL parameters captured by the [Router].
+  /// Returns `null` if no parameters are captured.
+  Map<String, String>? getParams(Request request) {
```

Comment:
If we decide to land this, how about the name `urlParameters`.

https://dart.dev/effective-dart/design#avoid-starting-a-method-name-with-get